authors/apps/tests/test_profile.py

```diff
@@ -37,3 +37,53 @@ def test_update_user_profile(self):
             format='json')
         self.assertEqual(response.data['bio'], 'coder')
         self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+
+class UserFollowingTestCase(BaseTestCase):
+
+    def test_follow_user(self):
+        self.register_user(new_user_2)
+        res = self.register_user(new_user)
+        self.add_credentials(res.data['token'])
+
+        response = self.client.post(
+            reverse('user_follow', kwargs={'username': 'John'}))
+        self.assertTrue(response.data['following'])
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_a_user_following_themselves(self):
+        res = self.register_user(new_user)
+        self.add_credentials(res.data['token'])
+
+        response = self.client.post(
+            reverse('user_follow', kwargs={'username': 'Jac'}))
+        self.assertIn('You can not follow yourself.',
+                      response.data['errors']['detail'])
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_unfollow_user(self):
+        self.register_user(new_user_2)
+        res = self.register_user(new_user)
+        self.add_credentials(res.data['token'])
+        self.client.post(reverse('user_follow', kwargs={'username': 'John'}))
+
+        response = self.client.delete(
+            reverse('user_follow', kwargs={'username': 'John'}))
+        self.assertFalse(response.data['following'])
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_follow_or_unfollow_user_who_does_not_exist(self):
+        res = self.register_user(new_user)
+        self.add_credentials(res.data['token'])
+
+        response = self.client.post(
+            reverse('user_follow', kwargs={'username': 'Joan'}))
+        self.assertIn(
+            'Profile does not exist. Cross-check the provided username.',
+            response.data['errors']['detail'])
+
+        response = self.client.delete(
+            reverse('user_follow', kwargs={'username': 'Joan'}))
+        self.assertIn(
+            'Profile does not exist. Cross-check the provided username.',
+            response.data['errors']['detail'])
```

Comment:
Well done Lule, but according to pep8 you need one empty line just below the last line

Reply:
It is there in the code, Github just doesn't include it in `Reader mode`.
![image](https://user-images.githubusercontent.com/36928255/49781867-536bef80-fd25-11e8-9e2c-07237c19a6fb.png)